components/schema-migrator/migrations/director/20211122114401_storage-redesign.up.sql

```diff
@@ -0,0 +1,714 @@
+BEGIN;
+
+CREATE TABLE tenant_applications
+(
+    tenant_id   uuid NOT NULL,
+    id uuid NOT NULL,
```

Comment:
Initially we started with the resource_id as the name of the column.

However, this does not work with the current way views are created. As can be seen below views are created in the form of `SELECT api.*, ta.tenant_id, ta.owner FROM ...`. Therefore the views has an id in them and the m2m tables has resource_id in them.

There are four ways to solve this that we evaluated:
1. Use id in the m2m tables to unify it with the views
2. Make a distinction in the code between tables and views and create different tenant isolation -> with `SELECT resource_id ...` when there is a m2m table and with `SELECT id ...` when there is a view.
3. Make the views exhaustive for all the columns (not an option since this will require recreating the views when new columns are added to the table).
4. Make the views with three columns only `SELECT id as resource_id, ta.tenant_id, ta.owner FROM ...`, but I think we had an idea to use those views in the ORD Service to read directly from them and we wanted all the columns in there.

We choose 1. as the easiest and least confusing option from our PoV. What do you think?